src/test/scala/chiselTests/ModulePrefixSpec.scala

```diff
@@ -357,4 +375,27 @@ class ModulePrefixSpec extends ChiselFlatSpec with ChiselRunners with Utils with
 
     matchesAndOmits(chirrtl)(lines: _*)()
   }
+
+  it should "omit the prefix if localModulePrefixUseSeparator is false" in {
+    class Foo extends RawModule {
+      override def localModulePrefix = Some("Inner")
+      override def localModulePrefixUseSeparator = false
+    }
+    class Top extends RawModule {
+      override def localModulePrefix = Some("Outer")
+      override def localModulePrefixUseSeparator = false
+      override def localModulePrefixAppliesToSelf = false
+      val foo = Module(new Foo)
+    }
+
+    val chirrtl = emitCHIRRTL(new Top)
+    val lines =
+      """
+      module OuterInnerFoo :
+      module Top :
+        inst foo of OuterInnerFoo
```

Comment:
I mean a test like `OuterMiddle_Inner_Foo`

Reply:
or maybe this test would be sufficient just put `Inner`s to `true`?

Reply:
This test is checking 2 different (but related) code paths, so I'll just add a separate test for your suggestion.